local_planner/src/nodes/histogram.cpp

```diff
@@ -48,59 +34,37 @@ void Histogram::downsample() {
   resolution = 2 * resolution;
   z_dim = z_dim / 2;
   e_dim = e_dim / 2;
-  std::vector<std::vector<double> > temp_bin;
-  std::vector<std::vector<double> > temp_age;
-  std::vector<std::vector<double> > temp_dist;
-  temp_bin.resize(e_dim);
-  temp_age.resize(e_dim);
-  temp_dist.resize(e_dim);
-  for (int i = 0; i < e_dim; ++i) {
-    temp_bin[i].resize(z_dim);
-    temp_age[i].resize(z_dim);
-    temp_dist[i].resize(z_dim);
-  }
+  Eigen::MatrixXd temp_age;
+  Eigen::MatrixXd temp_dist;
+  temp_age.resize(e_dim, z_dim);
+  temp_dist.resize(e_dim, z_dim);
+
   for (int i = 0; i < e_dim; ++i) {
     for (int j = 0; j < z_dim; ++j) {
       int i_high_res = 2 * i;
       int j_high_res = 2 * j;
 
-      double mean_bin =
-          (bin[i_high_res][j_high_res] + bin[i_high_res + 1][j_high_res] +
-           bin[i_high_res][j_high_res + 1] +
-           bin[i_high_res + 1][j_high_res + 1]) /
-          4.0;
       double mean_age =
-          (age[i_high_res][j_high_res] + age[i_high_res + 1][j_high_res] +
-           age[i_high_res][j_high_res + 1] +
-           age[i_high_res + 1][j_high_res + 1]) /
+          (age(i_high_res, j_high_res) + age(i_high_res + 1, j_high_res) +
+           age(i_high_res, j_high_res + 1) +
+           age(i_high_res + 1, j_high_res + 1)) /
           4.0;
       double mean_dist =
-          (dist[i_high_res][j_high_res] + dist[i_high_res + 1][j_high_res] +
-           dist[i_high_res][j_high_res + 1] +
-           dist[i_high_res + 1][j_high_res + 1]) /
+          (dist(i_high_res, j_high_res) + dist(i_high_res + 1, j_high_res) +
+           dist(i_high_res, j_high_res + 1) +
+           dist(i_high_res + 1, j_high_res + 1)) /
           4.0;
 
-      if (mean_bin >= 0.5) {
-        temp_bin[i][j] = 1.0;
-      } else {
-        temp_bin[i][j] = 0.0;
-      }
-      temp_age[i][j] = mean_age;
-      temp_dist[i][j] = mean_dist;
+      temp_age(i, j) = mean_age;
+      temp_dist(i, j) = mean_dist;
     }
   }
-  bin = temp_bin;
   age = temp_age;
```

Comment:
are these class members? Then they should be called `age_` and `dist_` instead. On this note, I'd also prefer them to be a bit more descriptive